libstorage/fs.c

```diff
@@ -74,7 +74,11 @@ void storage_fsHandler(void *data, msg_t *msg)
 			break;
 
 		case mtDevCtl:
-			msg->o.io.err = -EINVAL;
+			if (fs->ops->devctl == NULL) {
+				msg->o.io.err = -ENOSYS;
+				break;
+			}
+			(void)fs->ops->devctl(fs->info, &msg->i.io.oid, msg->i.raw, msg->o.raw);
```

Comment:
`devCtl`s can't return an error (I'm genuinely curious)?

Btw, `ioctl`s are `devCtl`s but they indeed pack `err` inside `o.raw` (they were implemented when there was no suitable `err` in msg) - maybe it could be straightened out in the near future ;)

Reply:
Oh, you're right, we should not set msg->o.io.err in this case.

Whether it can or not - we have no other way of passing this error than within msg->o.raw, so for now there's nothing we can do. We have to move err outside of msg->o.err and it is planned to be done.

I'm resolving this one then